src/koheesio/integrations/spark/tableau/hyper.py

```diff
@@ -0,0 +1,400 @@
+                    column_name = column.name.unescaped.__str__()
+                    tableau_type = column.type.__str__().lower()
+
+                    if tableau_type.startswith("numeric"):
+                        spark_type = DecimalType(precision=18, scale=5)
+                    else:
+                        spark_type = type_mapping.get(tableau_type, StringType)()
+
+                    if tableau_type == "timestamp" or tableau_type == "timestamp_tz":
+                        timestamp_cols.append(column_name)
+                        _col = f'cast("{column_name}" as text)'
+                    elif tableau_type == "date":
+                        date_cols.append(column_name)
+                        _col = f'cast("{column_name}" as text)'
+                    elif tableau_type.startswith("numeric"):
+                        _col = f'cast("{column_name}" as decimal(18,5))'
+                    else:
+                        _col = f'"{column_name}"'
+
+                    df_cols.append(StructField(column_name, spark_type))
+                    select_cols.append(_col)
+
+                data = connection.execute_list_query(f"select {','.join(select_cols)} from {self.table_name}")
+
+        df_schema = StructType(df_cols)
+        df = self.spark.createDataFrame(data, schema=df_schema)
+        if timestamp_cols:
+            df = CastToDatatype(column=timestamp_cols, datatype="timestamp").transform(df)
+        if date_cols:
+            df = CastToDatatype(column=date_cols, datatype="date").transform(df)
+
+        self.output.df = df
+
+
+class HyperFileWriter(HyperFile):
+    """
+    Base class for all HyperFileWriter classes
+    """
+
+    path: PurePath = Field(
+        default=TemporaryDirectory().name, description="Path to the Hyper file", examples=["PurePath(/tmp/hyper/)"]
+    )
+    name: str = Field(default="extract", description="Name of the Hyper file")
+    table_definition: TableDefinition = Field(
+        default=None,
+        description="Table definition to write to the Hyper file as described in "
+        "https://tableau.github.io/hyper-db/lang_docs/py/tableauhyperapi.html#tableauhyperapi.TableDefinition",
+    )
+
+    class Output(StepOutput):
+        """Output class for HyperFileListWriter"""
+
+        hyper_path: PurePath = Field(default=..., description="Path to created Hyper file")
+
+    @property
+    def hyper_path(self) -> Connection:
+        """
+        Return full path to the Hyper file.
+        """
+        if not os.path.exists(self.path):
+            os.makedirs(self.path)
+
+        hyper_path = PurePath(self.path, f"{self.name}.hyper" if ".hyper" not in self.name else self.name)
+        self.log.info(f"Destination file: {hyper_path}")
+        return hyper_path
+
+    def write(self):
+        self.execute()
+
+    @abstractmethod
+    def execute(self):
+        pass
+
+
+class HyperFileListWriter(HyperFileWriter):
+    """
+    Write list of rows to a Hyper file.
+
+    Reference:
```

Comment:
Please reformat this docstring to be Numpy compliant.

- Change headings to be like this
    > Reference
    > ----------
- Ensure the Example heading is plural (Examples)
- Put the code example in a code-block